src/components/cart/index.js

```diff
@@ -0,0 +1,144 @@
+          <a href="/payment">
+            <button class="os-btn-primary" data-element="orderBtn" data-cy="order-btn">Order</button>
+          </a>
+        </div>
+      </div>
+    `;
+  }
+
+  initEventListeners() {
+    // TODO: event "pointerdown" fails tests
+    this.element.addEventListener("click", (event) => {
+      const countBtn = event.target.closest("[data-counter]");
+
+      if (countBtn) {
+        const counterContainer = event.target.closest(
+          '[data-element="counterContainer"]',
+        );
+        const { id } = counterContainer.dataset;
+        const { counter } = countBtn.dataset;
+
+        if (counter === "1") {
+          this.dispatchEvent("increase-counter", id);
+        }
+
+        if (counter === "-1") {
+          this.dispatchEvent("decrease-counter", id);
+        }
+
+        const productCount = this.productStore.getProductCount(id);
+
+        if (productCount === 0) {
+          counterContainer.remove();
+        } else {
+          this.updatePrice(id);
+        }
+
+        this.updateTotal();
+      }
+    });
+
+    this.subElements.clearCartBtn.addEventListener("click", () => {
+      this.productStore.removeAll();
+      this.subElements.list.innerHTML = this.emptyCartMessage;
+      this.subElements.total.innerHTML = 0;
+    });
+  }
+
+  updatePrice(id) {
+    const product = this.productStore.get(id);
+    const counterContainer = this.element.querySelector(`[data-id="${id}"]`);
+
+    if (counterContainer) {
+      const price = counterContainer.querySelector('[data-element="price"]');
+      const counter = counterContainer.querySelector(`[data-element="${id}"]`);
+
+      price.innerHTML = product.price * product.count;
+      counter.innerHTML = product.count;
+    }
+  }
+
+  updateTotal() {
+    const total = this.productStore.getTotal();
+
+    this.subElements.total.innerHTML = total;
+
+    if (total === 0) {
+      this.subElements.list.innerHTML = this.emptyCartMessage;
+    }
+  }
+
+  createItem(item = {}) {
+    const wrapper = document.createElement("div");
+
+    const template = `
+      <li class="item-row" data-element="counterContainer" data-id="${item.id}">
+        <div class="item-preview">
+          <img src="${item.images[0]}" alt="${item.title}">
+        </div>
+        <div class="item-name">
+          ${item.title}
+        </div>
+        <div class="item-counter">
+          <button class="count-btn" data-counter="-1">
+            <i class="bi bi-dash-circle"></i>
+          </button>
+          <span data-element="${item.id}">${item.count}</span>
+          <button class="count-btn" data-counter="1">
+            <i class="bi bi-plus-circle"></i>
+          </button>
+        </div>
+        <div class="item-price" data-element="price">
+          ${item.price * item.count}
+        </div>
+      </li>
+    `;
+
+    wrapper.innerHTML = template;
+
+    return wrapper.firstElementChild;
+  }
```

Comment:
**Issue**: The method 'createItem' directly modifies innerHTML which can lead to cross-site scripting (XSS) vulnerabilities if not properly sanitized. <br> **Fix**: Ensure that any dynamic data inserted into the DOM via innerHTML is properly sanitized to prevent XSS attacks. <br> **Code Suggestion**: 
 ```
 - wrapper.innerHTML = template;
 + wrapper.textContent = sanitize(template); // Ensure sanitize function is implemented to clean input data
 ```